exonum-java-binding/core/rust/src/testkit/mod.rs

```diff
@@ -95,16 +96,14 @@ pub extern "system" fn Java_com_exonum_binding_testkit_TestKit_nativeCreateTestK
                 builder = builder.with_instance(instance);
             }
 
-            if let Some(service) =
+            if let Some(params) =
                 time_service_instance_from_java(&env, executor.clone(), time_service_spec)?
             {
-                // We always have exactly one time service instance.
-                let instance = service.instances[0].clone();
-                let artifact_id = service.factory.artifact_id();
+                let artifact_id = params.factory.artifact_id();
                 builder = builder
-                    .with_rust_service(service.factory)
                     .with_artifact(artifact_id)
-                    .with_instance(instance);
+                    .with_instance(&params)
+                    .with_rust_service(params.factory);
```

Comment:
It is very weird: Why doesn't testkit accept service factory in `with_artifact`?
Why on top of artifact and instance definition a factory is added with a _separate_
and _deceivingly named_ method (it says 'service' but accepts a factory of services — why isn't it passed with artifact)?